library/core/src/slice/mod.rs

```diff
@@ -2629,8 +2629,8 @@ impl<T> [T] {
     /// unstable (i.e. any number of equal elements may end up at position `index`), in-place
     /// (i.e. does not allocate), and *O*(*n*) worst-case. This function is also/ known as "kth
     /// element" in other libraries. It returns a triplet of the following values: all elements less
-    /// than the one at the given index, the value at the given index, and all elements greater than
-    /// the one at the given index.
+    /// than or equal to the one at the given index, the value at the given index, and all elements
+    /// greater than or equal the one at the given index.
```

Comment:
This isn't correct.  Rather than being those elements having a particular value, the resulting triplet is all those elements who, in the reordered slice, are respectively positioned before, at, and after `index`: only in the event that the element at `index` is unique can we then say with certainty that the triplet will contain those elements with values less than, equal to, and greater than that at `index`; if (as in the reporting issue) the value at `index` is non-unique, then its duplicates could sit in either or (as in the reporting issue) both of the "pre" and "post" subslices.